File: cpp/1023.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main()
{
    int c = 0;

    while (true) {
        int cons[201] = {0};
        int n, x, y, i, m;
        long long int s = 0, l = 0;
        cin >> n;

        if (n == 0) {
            break;
        }

        if (c > 0) {
            cout << '\n';
        }

        for (n; n > 0; n--) {
            cin >> x >> y;
            cons[y / x] += x;
            s += y;
            l += x;
        }

        for (i = 0; i < 201; i++) {
            if (cons[i] > 0) {
                m = i;
            }
        }

        c++;
        cout << "Cidade# " << c << ":\n";

        for (i = 0; i < m; i++) {
            if (cons[i] > 0) {
                cout << cons[i] << '-' << i << ' ';
            }
        }

        cout << cons[i] << '-' << i << '\n';
        printf("Consumo medio: %.2f m3.\n", ((100 * s) / l) / 100.0);
    }

    return 0;
}

File: py/1023.py
from math import floor

def drought():
    c = 0
    n = int(input())
    flag = False

    while n != 0:
        cons = dict()
        s = 0
        l = 0
        r = ''

        if flag:
            print('')

        for _ in range(n):
            try:
                x, y = [int(i) for i in input().split()]
                s += y
                l += x
                cons[y // x] += x

            except KeyError:
                cons[y // x] = x

        for i in range(201):
            try:
                r = "%s %s-%s" % (r, cons[i], i)

            except KeyError:
                continue

        c += 1
        print(f'Cidade# {c}:\n{r.lstrip()}')
        print(f'Consumo medio: {(floor((100 * s) / l) / 100):.2f} m3.')
        flag = True
        n = int(input())


if __name__ == "__main__":
    drought()